include/z64save.h

```diff
@@ -252,6 +252,16 @@ typedef enum {
     /* 3 */ GAMEMODE_END_CREDITS
 } GameMode;
 
+typedef enum {
+    /* 0 */ SCENE_LAYER_CHILD_DAY,
+    /* 1 */ SCENE_LAYER_CHILD_NIGHT,
+    /* 2 */ SCENE_LAYER_ADULT_DAY,
+    /* 3 */ SCENE_LAYER_ADULT_NIGHT    
+} SceneLayer;
+
+#define SCENE_LAYER_CUTSCENE_START (SCENE_LAYER_ADULT_NIGHT + 1)
```

Comment:
I'm unsure if this is better over
```suggestion
#define SCENE_LAYER_CUTSCENE_START 4
```
at all

Also I need to think more about it but I may like `SCENE_LAYER_CUTSCENE_FIRST` more, may be dumb but "cutscene start" sounds too much like "start a cutscene", but again idk

Reply:
I think its worth it from a shiftability point of view, if someone wants to add more "base/non cutscene" layers, for whatever reason
not saying its likely, but may as well make it less hardcoded

I am leaning toward just putting it in the enum though, sort of like the conventional "_MAX" entries, but it has a slightly different purpose

Reply:
yeah my point is i don't think it's more shiftable as `(SCENE_LAYER_ADULT_NIGHT + 1)` because entries past adult_night wouldn't shift the value. so this may seem shiftable when it's not
I'm not a fan of putting it in the enum but maybe it would be fine since the enum documents general layer behaviour / expectations. And it would bury this issue for good

Reply:
To add to that, there is a use case for having a proper `SCENE_LAYER_MAX = 21` for `SCENE_LAYER, since there's a special cutscene index (0xFFFD) used for certain cutscenes that don't require a "scene layer"

Reply:
I dont think it makes sense to define a value like that in terms of layers when it doesnt really involve them at all. Its a magic number as far as im concerned, just exists to satisfy the `gSaveContext.cutsceneIndex >= 0xFFF0` check for a cutscene script to be processed

Reply:
it's also one more than the last valid cutscene index, and is special cased to not play a cutscene if the value is set.